### PR TITLE
#184581333  ft(authorization):Protecting routes 

### PR DESCRIPTION
**Pull request by Patienceineza**
#### What does this PR do?
this  PR is for authentication or protection of routes 
#### Description of Task to be completed?

this task is associated with creating a middleware that will check if there is token in bearer auth and return an error otherwise it will verify that token and  either give access to user or deny access

#### How should this be manually tested?

-  after cloning the repo `write npm run drop` to drop previous tables and database

-  run `npm run migrate:all ` to run seeds and migrations
- and finally  run` npm install ` to install all necessary packages
- then go to where you test your api and use the correct End point to make sure it orks

#### Any background context you want to provide?

N/a

#### What are the relevant pivotal tracker/Trello stories?

https://www.pivotaltracker.com/story/show/184581333

### CODE REVIEW

**Review by nshutijonathan — CHANGES_REQUESTED** [review]
- Rename the title of your PR, It should not be like that 
- Rename your commit message from `Feature(Authorization):Protecting routes #184581333` to `feat(authorization):protecting routes` .

**Review by edwinnambaje — APPROVED** [review]
LGTM

**Review by Zeden12 — APPROVED** [review]
LGTM

**Review by blackd44 — APPROVED** [review]

File: src/controllers/users.controllers.js [hunk 1]
```diff
@@ -75,6 +75,7 @@ export default class Users {
     user.verified = true;
     await user.save();
     res.status(200).send({ message: 'Your account has been verified successfully!', verified: true });
+    console.log('User Verified Successfully');
```
Comment (nshutijonathan) [inline src/controllers/users.controllers.js]: remove `console.log`

File: src/routes/auth.routes.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import jwt from 'jsonwebtoken';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import bcrypt from 'bcrypt';
+import db from '../database/models';
+
+const router = express.Router();
+
+// eslint-disable-next-line operator-linebreak
+const JWT_SECRET =
+  'goK!pusp6ThEdURUtRenOwUhAsWUCLheBazl!uJLPlS8EbreWLdrupIwabRAsiBu';
```
Comment (nshutijonathan) [inline src/routes/auth.routes.js]: Hide the token, don't pass it here, It should be loaded from the `.env` file

**Review by nshutijonathan — CHANGES_REQUESTED** [review]

**Review by nshutijonathan — APPROVED** [review]
LGTM

**Pull request merged**